addons/account/models/sequence_mixin.py

```diff
@@ -169,7 +169,11 @@ def _get_last_sequence(self, relaxed=False):
         self.ensure_one()
         if self._sequence_field not in self._fields or not self._fields[self._sequence_field].store:
             raise ValidationError(_('%s is not a stored field', self._sequence_field))
-        where_string, param = self._get_last_sequence_domain(relaxed)
+        # (?P<no_reset>) can be used on the journal regex override to not reset the
+        # sequence on change of period, in that case all the sequences point to the
+        # journal override, so it's OK to only check with the monthly one
+        no_reset = "(?P<no_reset>)" in self._sequence_monthly_regex
+        where_string, param = self._get_last_sequence_domain(relaxed=(no_reset or relaxed))
```

Comment:
The mixin is not linked to journal entries and it is not supposed to be aware that there is a regex override. Please rework the comment taking that into account.

This should be in `_set_next_sequence`, it will allow you to set the month and year of the sequence correctly.

Reply:
Since IMO this is a hack to support the no reset, not a proper solution, I wanted to keep changes to a minimum.

Comments regarding `_set_next_sequence`, according to my understanding of it:

* We cannot rely on the `new` flag since we need to support migration from dbs where there are already moves that don't reset, which won't be covered by the `new` flag.
* The only way I see to pinpoint the right regex to check is on `_get_sequence_format_param` BUT this depends on the `last_sequence` that in turns depends on the `new` flag.
* The way I found to hijack the `last_sequence` was to set the `relaxed` flag if the group `no_reset` is found.

In general, I don't see a way to check for the `no_reset` group on the specific regex that matches the best. Thus I resort to checking on the monthly regex, since it is the first one checked on deduction of the format.